File: typescript/design-patterns/bridge/naive/naive-remotes.ts
// Naive approach: creating classes for every combination
class BasicTvRemote {
  turnOn() { console.log("TV on"); }
  setVolume(v: number) { console.log("TV volume to " + v); }
}

class AdvancedTvRemote extends BasicTvRemote {
  mute() { console.log("TV muted"); }
}

class BasicRadioRemote {
  turnOn() { console.log("Radio on"); }
  setVolume(v: number) { console.log("Radio volume to " + v); }
}

class AdvancedRadioRemote extends BasicRadioRemote {
  mute() { console.log("Radio muted"); }
}
// Adding a Projector -> creates 2 new classes.
// Adding a TouchPadRemote -> creates N new classes. Class explosion!

File: python/design-patterns/bridge/naive/naive_remotes.py
class BasicTvRemote:
    def turn_on(self): print("TV on")
    def set_volume(self, v): print(f"TV volume to {v}")

class AdvancedTvRemote(BasicTvRemote):
    def mute(self): print("TV muted")

class BasicRadioRemote:
    def turn_on(self): print("Radio on")
    def set_volume(self, v): print(f"Radio volume to {v}")

class AdvancedRadioRemote(BasicRadioRemote):
    def mute(self): print("Radio muted")